src/main/java/com/synopsys/integration/alert/web/security/authentication/event/AuthenticationEventListener.java

```diff
@@ -51,9 +50,12 @@ public String getDestinationName() {
     @Override
     public void handleEvent(AlertAuthenticationEvent event) {
         UserModel user = event.getUser();
-        Optional<UserModel> dbUser = userAccessor.getUser(user.getName());
-        if (dbUser.isEmpty()) {
-            userAccessor.addOrUpdateUser(user, true);
+        if (null != user) {
+            try {
+                userAccessor.addUser(user, true);
+            } catch (AlertDatabaseConstraintException ignored) {
+                // User already exists. Nothing to do.
```

Comment:
It's strange that we go through the effort of generating this exception for addUser then completely ignore it when it occurs. Since there's already a constraint on the DB level in regards to this, can we get away with not forcing this exception and perhaps adding a does user exist check when necessary?

Reply:
Isn't this for synching up users from LDAP or SAML.  So if the user has already logged in and the user exists we don't care.  But when we add a user through the API we will want the exception thrown there.  I think that's why this is ignored here.

Reply:
If you look at the original code, we attempt to get the user and, if absent, add the user to the DB. In the case where we added a new user, there would be two overlapping database api calls, now there is only one.